include/rocksdb/utilities/object_registry.h

```diff
@@ -13,17 +13,34 @@
 #include <string>
 #include <unordered_map>
 #include <vector>
+
+#include "rocksdb/configurable.h"
+#include "rocksdb/customizable.h"
 #include "rocksdb/status.h"
 
 namespace ROCKSDB_NAMESPACE {
+class DynamicLibrary;
+class Env;
 class Logger;
+class ObjectLibrary;
+class ObjectRegistry;
+struct ColumnFamilyOptions;
+struct DBOptions;
+
 // Returns a new T when called with a string. Populates the std::unique_ptr
 // argument if granting ownership to caller.
 template <typename T>
 using FactoryFunc =
     std::function<T*(const std::string&, std::unique_ptr<T>*, std::string*)>;
 
-class ObjectLibrary {
+// The signature of the function for loading factories
+// into an object library.  This method is expected to register
+// factory functions in the supplied ObjectLibrary.
+// @param library   The library to load factories into.
+// @param arg       Argument to the library loader
+using RegistrarFunc = std::function<void(ObjectLibrary&, const std::string&)>;
+
+class ObjectLibrary : public Customizable {
```

Comment:
Add some comments to explain 1) what is ObejctLibrary, 2) why it is inherited from Customizable.

Reply:
This is not part of the current submission.  The PR was broken into more pieces and this will be in a follow-on